### DISCUSSION TITLE
Clear button for AutoComplete

### DISCUSSION THREAD

**Discussion by tmlmt** [primefaces/community#437] [PrimeVue]
Reposting here the issue posted by @cuberinooo not so long aso (https://github.com/primefaces/primevue/issues/4108) requesting a "Clear input" button for AutoComplete. It would be a great idea. Any others agreeing with this?

**Comment by tmlmt** [comment]
Workaround: 
```js
<span class="p-input-icon-right">
  <AutoComplete ...>
  <i
    v-if="value.length > 0"
    class="pi pi-times-circle"
    style="cursor: pointer"
    @click="value = ''"
  />
</span>
```

**Reply by mjbates7** [reply]
Thank you, hopefully add to the component via slot or prop

**Comment by mjbates7** [comment]
Also looking for this. Would be a good addition to add a prop or slot to allow the input to be cleared once a selection has been made. I'm thinking it can go on the right hand side by default (where the loading icon goes)

**Comment by yureckey** [comment]
Thanks for workaround, but I agree - it would be good to have this built-in for all the different inputs

**Comment by Tarabass** [comment]
I was looking for this. In my opinion this should be configurable like `show-clear` on `input text`.

**Comment by Acetyld** [comment]
Also agree 👏💪🏻

**Comment by ben-hamel** [comment]
agreed, would be great to have this added

**Comment by EmilMoe** [comment]
I was coming here to suggest the exact same. If it can work as the regular `select` I'm very pleased. 👍

**Comment by anchan42** [comment]
+1
